src/main/java/com/conveyal/taui/controllers/BundleController.java

```diff
@@ -71,16 +63,9 @@ public static Bundle create (Request req, Response res) {
 
         Persistence.bundles.create(bundle);
 
-        File bundleFile = null;
         File directory = Files.createTempDir();
         List<File> localFiles = new ArrayList<>();
         try {
-            // cache bundle on disk to avoid OOME
```

Comment:
We need to add a lot of comments to this class and method. I just spent quite a long time trying to understand what it's for and how it works. First, the use of the word "Bundle" here needs explanation. It's creating a record in the application database, but also files on the local drive, and also an assemblage of those files that are uploaded via a cache. But the unstated fact is that it's processing a GTFS file upload. 

It creates a `File bundleFile` but that only ever seems to be deleted, not used for anything. This seems to be followed by our own hand-rolled code to make unique filenames (I'm not sure why we are not using a library function for this). It looks like all the uploaded files are copied into a temporary directory, where they are processed one by one.

I'm still trying to figure out where these files get uploaded to S3 in the original code. I only see it uploading the bundle "manifest" but not the bundle files themselves. I'm looking because I'm concerned that in local operation, we'll be "uploading" files via the browser to a temporary file, then that temporary file will then be "uploaded" to the local file storage (swapped in instead of S3 storage). It is possible that the files uploaded from the UI are already stored in temporary files, or (worse?) in memory, before we manually copy them to our own temporary files. This will yield three to four copies of a file that is then treated as read-only, possibly keeping it entirely in memory for some duration. This is problematic for files upward of 500MB or 1GB, and something we need to understand and solve in our storage design. It may be responsible for some instability we've seen in the past.

I have not yet verified this definitively, but it looks like ServletFileUpload is using org.apache.commons.fileupload.disk.DiskFileItem, which actually flushes to disk above 10MB (see org.apache.commons.fileupload.disk.DiskFileItemFactory#DEFAULT_SIZE_THRESHOLD) so the uploaded files are already temp files, we're moving them to new temp files with our own generated names, then will be copying those files to the LocalFilePersistence. 

Aha, the call chain that gets each feed into the storage is ApiMain.registerFeedSource() -> FeedSourceCache.put() -> BaseGTFSCache.retrieveAndProcessFeed(). So this is going into the old MapDB backed gtfs-lib, which has its own S3 magic.

All details and design decisions around things like this should be mentioned in comments - it could save us hours or days of work in the future.